Camera2SlowMotion/app/src/main/java/com/example/android/camera2/slowmo/MainActivity.java

```diff
@@ -188,63 +165,53 @@ public void onSuccess(Text visionText) {
+                        setFrameTimeStamp();
                         framesProcessed++;
                         if (framesProcessed == frameList.size()) {
                           analyseResultField.append("OCR on all frames done\n");
-                          parseOCR();
-                          analyseResultField.append("OCR frames parsed\n");
-                          showResults();
+                          calculateLag();
                         }
                       }
                     })
                 .addOnFailureListener(
                     new OnFailureListener() {
                       @Override
                       public void onFailure(@NonNull Exception e) {
-                        Log.d("Rokus Logs:", "Analyse failed with: " + e.getMessage());
+                        Log.d(TAG_AUTHOR, "Analyse failed with: " + e.getMessage());
                       }
                     });
       }
       return null;
     }
   }
 
-  private void parseOCR() {
+  private void setFrameTimeStamp() {
     try {
-      Long recordStartTime = CameraFragment.Companion.getRecordingStartMillis();
-      Long offset = 1000L / fps;
-      for (int i = 0; i < resultsOCR.size(); i++) {
-        deltaOCR.add(recordStartTime + (i * offset));
-      }
+      videoFrameTimestamp.add(recordStartTime + (framesProcessed * frameDuration));
     } catch (Exception e) {
-      Log.d("Rokus Logs:", "Parsing OCR failed with: " + e.getMessage());
-    }
-  }
-
-  // Parse timestamps received from server.
-  private void parseServer() {
-    for (int i = 0; i < serverCache.size(); i++) {
-      deltaServer.add(Long.valueOf(serverCache.get(i)));
-      Log.d(tag, "ParseServer:" + deltaServer.get(deltaServer.size() - 1));
+      videoFrameTimestamp.add(0L);
+      Log.d(TAG_AUTHOR, "Unable to set timeStamp of frame= " + framesProcessed);
+      Log.d(TAG_AUTHOR, e.getMessage());
     }
   }
 
-  private void showResults() {
-    if (deltaServer.size() < 2 || deltaOCR.size() < 2) {
-      Log.d(tag, "No results to show");
+  private void calculateLag() {
+    if (serverTimestamp.size() < 2 || videoFrameTimestamp.size() < 2) {
+      Log.d(TAG_AUTHOR, "No results to show");
       return;
     }
     Log.d(
         "Show Results",
-        "Video start timestamp:" + CameraFragment.Companion.getRecordingStartTime());
+        "Video start timestamp:" + recordStartTime);
+    //TODO: Server sequence is hardcoded. Add functionality to receive serverSequence.
     String serverSequence = "m";
-    sync_offset = deltaServer.get(0) - CameraFragment.Companion.getRecordingStartMillis();
-    Log.d("Show Results", "Sync offset:" + sync_offset);
-    for (int j = 1; j < deltaServer.size(); j++) {
+    syncOffset = serverTimestamp.get(0) - CameraFragment.Companion.getRecordingStartMillis();
+    Log.d("Show Results", "Sync offset:" + syncOffset);
+    for (int j = 1; j < serverTimestamp.size(); j++) {
```

Comment:
I feel like this timestamp data and related calculations should be moved to a separate class so that it can be grouped together more obviously and not distributed through all of the activity and task scheduling code in this class.